lib/pf/Switch/HP/Procurve_2920.pm

```diff
@@ -0,0 +1,145 @@
+package pf::Switch::HP::Procurve_2920;
+
+=head1 NAME
+
+pf::Switch::HP::Procurve_2920 - Object oriented module to access SNMP enabled HP Procurve 2920 switches
+
+=head1 SYNOPSIS
+
+The pf::Switch::HP::Procurve_2920 module implements an object 
+oriented interface to access SNMP enabled HP Procurve 2920 switches.
+
+=head1 BUGS AND LIMITATIONS
+
+VoIP not tested using MAC Authentication/802.1X
+
+=cut
+use strict;
+use warnings;
+use Log::Log4perl;
+use Net::SNMP;
+use base ('pf::Switch::HP');
+
+sub description { 'HP ProCurve 2920 Series' }
+
+# importing switch constants
+use pf::Switch::constants;
+use pf::util;
+use pf::config;
+
+# CAPABILITIES
+# access technology supported
+sub supportsWiredMacAuth { return $TRUE; }
+sub supportsWiredDot1x { return $TRUE; }
+sub supportsLldp { return $TRUE; }
+sub supportsRadiusVoip { return $TRUE; }
+# inline capabilities
+sub inlineCapabilities { return ($MAC,$PORT); }
+
+=item getVoipVSA
+Get Voice over IP RADIUS Vendor Specific Attribute (VSA).
+=cut
+sub getVoipVsa {
+        my ($this) = @_;
+        my $logger = Log::Log4perl::get_logger(ref($this));
+        my $vlanid = sprintf("%03x\n",$this->getVlanByName('voice'));
+        my $hexvlan = hex("31000".$vlanid);
+        return ('Egress-VLANID' => $hexvlan,);
+}
+
+
+=item isVoIPEnabled
+Supports VoIP if enabled.
+=cut
+sub isVoIPEnabled {
+        my ($self) = @_;
+        return ( $self->{_VoIPEnabled} == 1 );
+}
+
+=item getPhonesLLDPAtIfIndex
+Using SNMP and LLDP we determine if there is VoIP connected on the switch port
+=cut
+sub getPhonesLLDPAtIfIndex {
+        my ( $this, $ifIndex ) = @_;
+        my $logger = Log::Log4perl::get_logger( ref($this) );
+        my @phones;
+        if ( !$this->isVoIPEnabled() ) {
+                $logger->debug( "VoIP not enabled on switch "
+                        . $this->{_ip}
+                        . ". getPhonesLLDPAtIfIndex will return empty list." );
+                return @phones;
+        }
+        my $oid_lldpRemPortId = '1.0.8802.1.1.2.1.4.1.1.7';
+        my $oid_lldpRemChassisIdSubtype = '1.0.8802.1.1.2.1.4.1.1.4';
+        if ( !$this->connectRead() ) {
+                return @phones;
+        }
+        sleep(4);
```

Comment:
Is this really necessary ? If yes add a small comment above for which value you're waiting for